src/app/creator-profile-page/creator-profile-nfts/creator-profile-nfts.component.ts

```diff
@@ -125,70 +126,77 @@ export class CreatorProfileNftsComponent implements OnInit {
+              bidEntry.PostEntryResponse = res.PostHashHexToPostEntryResponse[bidEntry.PostHashHex];
+              return bidEntry;
+            });
+            this.lastPage = Math.floor(this.myBids.length / CreatorProfileNftsComponent.PAGE_SIZE);
+            const sortDetails = this.sortFields[this.orderNFTsBy];
+            return _.orderBy(this.myBids, [sortDetails.field], [sortDetails.order]);
+          }
+        )
       );
   }
 
-  getNFTs(isForSale: boolean | null = null, isPending: boolean | null = null): Subscription {
+  getNFTs(isForSale: boolean | null = null, isPending: boolean | null = null) {
     return this.backendApi
       .GetNFTsForUser(
         this.profile.PublicKeyBase58Check,
         this.globalVars.loggedInUser?.PublicKeyBase58Check,
         isForSale,
         isPending
       )
-      .subscribe(
-        (res: {
-          NFTsMap: { [k: string]: { PostEntryResponse: PostEntryResponse; NFTEntryResponses: NFTEntryResponse[] } };
-        }) => {
-          this.nftResponse = [];
-          for (const k in res.NFTsMap) {
-            const responseElement = res.NFTsMap[k];
-            // Exclude NFTs created by profile from Gallery
-            if (
-              this.activeTab === CreatorProfileNftsComponent.MY_GALLERY &&
-              responseElement.PostEntryResponse.PosterPublicKeyBase58Check === this.profile.PublicKeyBase58Check
-            ) {
-              continue;
+      .pipe(
+        tap(
+          (res: {
+            NFTsMap: { [k: string]: { PostEntryResponse: PostEntryResponse; NFTEntryResponses: NFTEntryResponse[] } };
+          }) => {
+            this.nftResponse = [];
+            for (const k in res.NFTsMap) {
+              const responseElement = res.NFTsMap[k];
+              // Exclude NFTs created by profile from Gallery
+              if (
+                this.activeTab === CreatorProfileNftsComponent.MY_GALLERY &&
+                responseElement.PostEntryResponse.PosterPublicKeyBase58Check === this.profile.PublicKeyBase58Check
+              ) {
+                continue;
+              }
+              this.nftResponse.push(responseElement);
             }
-            this.nftResponse.push(responseElement);
+            this.lastPage = Math.floor(this.nftResponse.length / CreatorProfileNftsComponent.PAGE_SIZE);
+            const sortDetails = this.sortFields[this.orderNFTsBy];
+            return _.orderBy(this.nftResponse, [sortDetails.field], [sortDetails.order]);
           }
-          this.lastPage = Math.floor(this.nftResponse.length / CreatorProfileNftsComponent.PAGE_SIZE);
-          const sortDetails = this.sortFields[this.orderNFTsBy];
-          return _.orderBy(this.nftResponse, [sortDetails.field], [sortDetails.order]);
-        }
+        )
       );
   }
 
   getPage(page: number) {
-    if (this.lastPage != null && page > this.lastPage) {
+    if (this.lastPage !== null && page > this.lastPage) {
```

Comment:
is this a safe null check? I think maybe we should default to using the isNil helper from lodash for null comparisons just to be on the safe side and so there is no room for mistakes.